go/libraries/doltcore/merge/merge.go

```diff
@@ -859,7 +868,6 @@ func MergeRoots(ctx context.Context, ourRoot, theirRoot, ancRoot *doltdb.RootVal
 	// need to validate merges can be done on all tables before starting the actual merges.
 	for _, tblName := range tblNames {
```

Comment:
One lingering issue is we still match tables by name, rather than by tags, across merge roots. So renamed tables will not be handled correctly

Reply:
Added a note and TODO